### PR TITLE
Add org.kde.qmlkonsole

### PR DESCRIPTION
**Pull request by plata**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by plata** [comment]
CC @flathub/kde

**Comment by plata** [comment]
bot, build org.kde.telly-skout

**Comment by plata** [comment]
@espidev @JBBgameich is this ok for you?

**Comment by plata** [comment]
bot, build org.kde.qmlkonsole

**Comment by plata** [comment]
bot, build org.kde.qmlkonsole

**Comment by plata** [comment]
@Swordfish90 any plans for a new QMLTermWidget release?

File: org.kde.qmlkonsole.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    "id": "org.kde.qmlkonsole",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15-22.08",
+    "sdk": "org.kde.Sdk",
+    "command": "qmlkonsole",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
```
Comment (hfiguiere) [inline org.kde.qmlkonsole.json:11]: it's either only `x11` or `fallback-x11` with wayland

File: org.kde.qmlkonsole.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        "--device=dri",
+        "--filesystem=home",
+        "--talk-name=org.freedesktop.Notifications",
+        "--talk-name=org.freedesktop.Flatpak"
```
Comment (hfiguiere) [inline org.kde.qmlkonsole.json:15]: this dbus permission is why the build fails.

File: org.kde.qmlkonsole.json [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                "-DBUILD_TESTING=OFF"
+            ]
+        },
+        {
+            "name": "qmltermwidget",
+            "buildsystem": "qmake",
+            "make-install-args": [ "INSTALL_ROOT=/app" ],
+            "post-install": [
+                "mkdir -p /app/lib/qml",
+                "ln -s /app/usr/lib/qml/QMLTermWidget /app/lib/qml",
```
Comment (hfiguiere) [inline org.kde.qmlkonsole.json:46]: I'm pretty sure the problem here is that PREFIX isn't followed.

**Comment by espidev** [comment]
> @espidev @JBBgameich is this ok for you?

With the Flatpak I struggled with figuring out the permissions to get the shell working, ex. `sudo`

File: org.kde.qmlkonsole.json (hunk 2 above)
Comment (JBBgameich) [reply org.kde.qmlkonsole.json:15]: There is no more fine-grained permission to allow spawning on the host, right? For a terminal app that is pretty much required to be useful.

File: org.kde.qmlkonsole.json (hunk 3 above)
Comment (JBBgameich) [reply org.kde.qmlkonsole.json:46]: Yes, qmake is by itself not able to install to a different prefix than the one the Qt install uses.
So I've seen this workaround in other flatpaks as well.

**Comment by JBBgameich** [comment]
This currently fails to build in the qmlkonsole module with
```
CMake Error at CMakeLists.txt:15 (message):
  This application requires an out of source build.  Please create a separate
  build directory.
```

We should probably get rid of that upstream, but in the meantime you can add `"builddir": true,`

File: org.kde.qmlkonsole.json [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    "runtime-version": "5.15-22.08",
+    "sdk": "org.kde.Sdk",
+    "command": "qmlkonsole",
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--filesystem=home",
```
Comment (JBBgameich) [inline org.kde.qmlkonsole.json:13]: Should not be necessary

File: org.kde.qmlkonsole.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        "--talk-name=org.freedesktop.Notifications",
```
Comment (JBBgameich) [inline org.kde.qmlkonsole.json:14]: This is currently unused as well I think

**Comment by JBBgameich** [comment]
< With the Flatpak I struggled with figuring out the permissions to get the shell working, ex. sudo

We currently use a pretty bad hack to get bash to work with flatpak-spawn at all. Using https://github.com/1player/host-spawn would make it a lot more complete and probably also make sudo work.

**Comment by plata** [comment]
Can you provide the patch file when you feel it's ready?

**Comment by JBBgameich** [comment]
I already opened a PR to your branch https://github.com/plata/flathub/pull/1

**Comment by plata** [comment]
> I already opened a PR to your branch [plata#1](https://github.com/plata/flathub/pull/1)

Sorry, I didn't see that (there was no notification).

**Comment by plata** [comment]
bot, build org.kde.qmlkonsole

**Comment by plata** [comment]
bot, build org.kde.qmlkonsole

**Comment by JBBgameich** [comment]
This needs an exception just like https://github.com/flathub/flatpak-builder-lint/blob/6c508cc4799830043008405bf1b74c575b2ea1f8/flatpak_builder_lint/staticfiles/exceptions.json#L1206

**Comment by plata** [comment]
bot, build org.kde.qmlkonsole

**Comment by plata** [comment]
bot, build org.kde.qmlkonsole

**Comment by JBBgameich** [comment]
Can you also submit the appdata changes upstream when they are ready?

The current build failure is a bit unclear for me too, in the past svg-icons definitely were enough.

**Comment by plata** [comment]
bot, build org.kde.qmlkonsole

**Comment by plata** [comment]
Any ideas for this error? Looks more like something in the CI to me.

**Comment by plata** [comment]
bot, build org.kde.qmlkonsole

**Comment by plata** [comment]
Or is it missing a permission now?

**Comment by JBBgameich** [comment]
The error is 

> Error loading desktop file: Failed to find icon: Failed to find icon org.kde.qmlkonsole

But I don't understand what causes it right now

**Comment by plata** [comment]
Isn't that just the result of
```
fusermount3: failed to access mountpoint /run/user/996/doc: Permission denied
error: fuse init failed: Can't mount path /run/user/996/doc
```
?

**Comment by plata** [comment]
bot, build org.kde.qmlkonsole

**Comment by plata** [comment]
see https://github.com/flatpak/flatpak-builder/issues/228

**Comment by plata** [comment]
@espidev @JBBgameich can you answer the open topics in the PR description?

**Comment by JBBgameich** [comment]
> - [x] I am using only the minimal set of permissions. (If not, please explain each non-standard permission.)
> - [x] All assets referenced in the manifest are redistributable by any party. If not, the unredistributable parts are using an extra-data source type.
> - [x] I am an upstream contributor to the project.

**Pull request closed without merging**